src/Conjure/Language/NewLexer.hs

```diff
@@ -0,0 +1,309 @@
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE InstanceSigs #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE TypeFamilies #-}
+
+module Conjure.Language.NewLexer (
+    ETok(..),
+    Offsets(..),
+    Reformable(..),
+    prettySplitComments,
+    eLex,
+    reformList,
+    tokenSourcePos,
+    totalLength,
+    trueLength,
+    tokenStart,
+    sourcePos0,
+    LexerError(..),
+    runLexer,
+    ETokenStream(..)
+    ) where
+
+import Conjure.Language.Lexemes
+import Conjure.Prelude hiding (many, some,Text)
+import Data.Char (isAlpha, isAlphaNum)
+import Data.Void
+
+import qualified Data.Text as T
+import qualified Data.Text.Lazy as L
+import Data.Text (Text)
+import Text.Megaparsec hiding (State)
+import Text.Megaparsec.Char
+
+import Data.List (splitAt)
+import qualified Data.List.NonEmpty as NE
+import qualified Text.Megaparsec as L
+import Prelude (read)
+import  Prettyprinter as Pr
+
+
+sourcePos0 :: SourcePos
+sourcePos0 = SourcePos "" (mkPos  1) (mkPos 1)
+
+class Reformable a where
+    reform :: a -> L.Text
+
+instance Reformable ETok where
+    reform e | oTLength (offsets e) == 0 = ""
+    reform (ETok{capture=cap,trivia=triv}) = L.append  (L.concat $ map showTrivia triv) (L.fromStrict cap)
+        where
+            showTrivia :: Trivia -> L.Text
+            showTrivia x = case x of
+              WhiteSpace txt    -> L.fromStrict txt
+              LineComment txt   -> L.fromStrict txt
+              BlockComment txt  -> L.fromStrict txt
+
+reformList :: (Traversable t ,Reformable a) => t a -> L.Text
+reformList =  L.concat  . map reform . toList
+
+emojis :: [Char]
```

Comment:
Oz to check why emojis don't work. Add the following as a test case

https://conjure.readthedocs.io/en/latest/tutorials.html#bibd

Reply:
This should now be fixed